plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/internal/CloudSdkModifyJob.java

```diff
@@ -54,11 +57,22 @@
   public CloudSdkModifyJob(String jobName, MessageConsoleStream consoleStream,
       ReadWriteLock cloudSdkLock) {
     super(jobName);
-    this.consoleStream = consoleStream;
+    this.consoleStream = consoleStream != null ? consoleStream : createNewMessageConsole();
     this.cloudSdkLock = cloudSdkLock;
     setRule(MUTEX_RULE);
   }
 
+  @VisibleForTesting
+  MessageConsoleStream createNewMessageConsole() {
+    MessageConsole console = MessageConsoleUtilities.findOrCreateConsole(
+        Messages.getString("configuring.cloud.sdk"), // $NON-NLS-1$
+        name -> new MessageConsole(name, null /* imageDescriptor */));
```

Comment:
Use of lambda here feels a little gratuitous. What do we gain by not simply passing in the name ?

Reply:
The API accepts a factory class. In this case, we don't need a custom console creator, but others like the deploy console, they need to create and return customized consoles. Maybe we could add a new API that just accepts a name. Let me look into it.

Reply:
Actually, there did exist another method that just accepts a name. It works well.